src/common/connection.js

```diff
@@ -176,7 +176,8 @@ class Connection extends EventEmitter {
         this._ws.on('error', error =>
           this.emit('error', 'websocket', error.messsage, error));
         this._ws.on('message', this._onMessage.bind(this));
-        this._onUnexpectedCloseBound = this._onUnexpectedClose.bind(this);
+        this._onUnexpectedCloseBound = this._onUnexpectedClose.bind(this,
+          resolve, reject);
```

Comment:
Fix it here 
https://github.com/ripple/ripple-lib/pull/686/files#diff-c5fa987b0954acfe830ac6adda20f19fR106